packages/core/src/useSyncState.tsx

```diff
@@ -1,74 +1,46 @@
 import * as React from 'react';
 
-const UNINTIALIZED_STATE = {};
+const createStore = <T,>(getInitialState: () => T) => {
+  const listeners: (() => void)[] = [];
 
-/**
- * This is definitely not compatible with concurrent mode, but we don't have a solution for sync state yet.
- */
-export function useSyncState<T>(initialState?: (() => T) | T) {
-  const stateRef = React.useRef<T>(UNINTIALIZED_STATE as any);
-  const isSchedulingRef = React.useRef(false);
-  const isMountedRef = React.useRef(true);
+  let state: T = getInitialState();
 
-  React.useEffect(() => {
-    isMountedRef.current = true;
+  const getState = () => state;
 
-    return () => {
-      isMountedRef.current = false;
-    };
-  }, []);
-
-  if (stateRef.current === UNINTIALIZED_STATE) {
-    stateRef.current =
-      // @ts-expect-error: initialState is a function, but TypeScript doesn't think so
-      typeof initialState === 'function' ? initialState() : initialState;
-  }
-
-  const [trackingState, setTrackingState] = React.useState(stateRef.current);
-
-  const getState = React.useCallback(() => stateRef.current, []);
+  const setState = (newState: T) => {
+    state = newState;
+    listeners.forEach((listener) => listener());
+  };
 
-  const setState = React.useCallback((state: T) => {
-    if (state === stateRef.current || !isMountedRef.current) {
-      return;
-    }
+  const subscribe = (callback: () => void) => {
+    listeners.push(callback);
```

Comment:
personally, I'd prefer using objects at this will be used quite a lot (a bit better access and removing time)

Reply:
Actually, it's used directly only once so w/e